src/index.js

```diff
@@ -39,6 +46,55 @@ class Offline {
     this.exitCode = 0;
     this.provider = 'aws';
     this.start = this.start.bind(this);
+    this.coverageDataOutputPath = process.env.SLS_OFFLINE_COVERAGE_DATA_OUTPUT_PATH;
+    this.shouldOutputCoverage = typeof this.coverageDataOutputPath !== 'undefined';
+
+    if (this.shouldOutputCoverage) {
+      debugLog(
+        `Code coverage will be written to ${this.coverageDataOutputPath}`
+      );
+      /* eslint-disable new-cap */
+      this.coverageQueue = new pQueue({ concurrency: 1 });
+      /* eslint-enable new-cap */
+      this.outputCoverage = pDebounce(
```

Comment:
The coverage output code is [debounced](https://css-tricks.com/debouncing-throttling-explained-examples/#article-header-id-0) so it doesn't spam I/O while tests are being executed in rapid succession. The queue ensures the output file is not written to multiple times at once, but now that I'm looking at this again I'm not sure the queue is necessary anymore.